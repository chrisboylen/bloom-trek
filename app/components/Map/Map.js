import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { MapView } from 'expo';

const deltas = {
	latitudeDelta: 0.0922,
	longitudeDelta: 0.0421
};

const initialRegion = {
	latitude: 39.739,
  longitude: -104.99,
};

const Marker = MapView.Marker;

export default class Map extends Component {

	renderMarkers() {
		return this.props.parks.map((park, i) => (
			<Marker key={i} title={park.name} coordinate={park.coords} />
		));
	}

	render() {
		const region = {
			...this.props.location,
      ...deltas
		};

		if (!region.latitude || !region.longitude) {
			return (
				<View>
					<Text>Loading map...</Text>
				</View>
			);
		}

		return (
			<MapView
				style={styles.container}
				region={region}
				initialRegion={{ ...initialRegion, ...deltas }}
        showsUserLocation
        showsMyLocationButton
			>
				{this.renderMarkers()}
			</MapView>
		);
	}
}

const styles = StyleSheet.create({
	container: {
		width: '100%',
		height: '100%'
	}
});
