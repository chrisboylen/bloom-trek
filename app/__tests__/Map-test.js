import React from 'react';
import { shallow, mount } from 'enzyme';
import { MapView } from 'expo';
import Map from '../components/Map';

describe('Map', () => {
  let wrapper;
  let mockHandleSelectedPark;
  let mockHandleSelectedTrail;
  let mockParks;
  let mockTrails;

  beforeEach(() => {
    mockHandleSelectedPark = jest.fn();
    mockHandleSelectedTrail = jest.fn();
    wrapper = shallow(
      <Map
        handleSelectedPark={mockHandleSelectedPark}
        handleSelectedTrail={mockHandleSelectedTrail}
      />,
    );
    mockParks = [{
      id: 1,
      name: 'Castlewood Canyon',
      coords: {
        latitude: 39.3379,
        longitude: -104.7512,
      },
    }];
    mockTrails = [{
      id: 1,
      name: 'Castlewood Canyon - Rimrock to Creek Bottom Loop',
      uid: 7006769,
      difficulty: 'medium',
      length: 4.6,
      status: 'All Clear',
      trail_img_url: 'https://cdn-files.apstatic.com/hike/7006325_medium_1438452681.jpg',
      trail_url: 'https://www.hikingproject.com/trail/7006769/castlewood-canyon-rimrock-to-creek-bottom-loop',
      summary: 'A fun, rolling trail with a sense of seclusion, great geology, and historical sites.',
      park_id: 1,
      coords: {
        latitude: 39.3598,
        longitude: -104.7682,
      },
    }];
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleParkMarkerPress', () => {
    it('should invoke handeSelectedPark when called', () => {
      wrapper.instance().handleParkMarkerPress();

      expect(mockHandleSelectedPark).toHaveBeenCalled();
    });
  });

  describe('HandleTrailMarkerPress', () => {
    it('should invoke handleSelectedTrail', () => {
      wrapper.instance().handleTrailMarkerPress();

      expect(mockHandleSelectedTrail).toHaveBeenCalled();
    });
  });
});
