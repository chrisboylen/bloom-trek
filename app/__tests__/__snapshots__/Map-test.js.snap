// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Map
    handleSelectedPark={[MockFunction]}
    handleSelectedTrail={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Loading map...
      </Text>,
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Loading map...",
        "ellipsizeMode": "tail",
      },
      "ref": null,
      "rendered": "Loading map...",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Loading map...
        </Text>,
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Loading map...",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "Loading map...",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
